wsagent/che-core-api-languageserver/src/main/java/org/eclipse/che/api/languageserver/InitializeParamsProvider.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.languageserver;
+
+import static org.eclipse.che.api.fs.server.WsPathUtils.ROOT;
+
+import com.google.inject.Singleton;
+import java.lang.management.ManagementFactory;
+import java.net.URI;
+import java.nio.file.Paths;
+import javax.inject.Inject;
+import org.eclipse.che.api.fs.server.PathTransformer;
+import org.eclipse.che.api.languageserver.RegistryContainer.Registry;
+import org.eclipse.lsp4j.ClientCapabilities;
+import org.eclipse.lsp4j.CodeActionCapabilities;
+import org.eclipse.lsp4j.CodeLensCapabilities;
+import org.eclipse.lsp4j.CompletionCapabilities;
+import org.eclipse.lsp4j.CompletionItemCapabilities;
+import org.eclipse.lsp4j.DefinitionCapabilities;
+import org.eclipse.lsp4j.DidChangeConfigurationCapabilities;
+import org.eclipse.lsp4j.DidChangeWatchedFilesCapabilities;
+import org.eclipse.lsp4j.DocumentHighlightCapabilities;
+import org.eclipse.lsp4j.DocumentLinkCapabilities;
+import org.eclipse.lsp4j.DocumentSymbolCapabilities;
+import org.eclipse.lsp4j.ExecuteCommandCapabilities;
+import org.eclipse.lsp4j.FormattingCapabilities;
+import org.eclipse.lsp4j.HoverCapabilities;
+import org.eclipse.lsp4j.InitializeParams;
+import org.eclipse.lsp4j.OnTypeFormattingCapabilities;
+import org.eclipse.lsp4j.RangeFormattingCapabilities;
+import org.eclipse.lsp4j.ReferencesCapabilities;
+import org.eclipse.lsp4j.RenameCapabilities;
+import org.eclipse.lsp4j.SignatureHelpCapabilities;
+import org.eclipse.lsp4j.SymbolCapabilities;
+import org.eclipse.lsp4j.SynchronizationCapabilities;
+import org.eclipse.lsp4j.TextDocumentClientCapabilities;
+import org.eclipse.lsp4j.WorkspaceClientCapabilities;
+import org.eclipse.lsp4j.WorkspaceEditCapabilities;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Utility class that simplifies creation of a language server initialize parameters. */
+@Singleton
+class InitializeParamsProvider {
```

Comment:
Might be quite true. I guess this and some other names should be reviewed.

Reply:
After a certain thought I prefer to leave it as is, because again factories limit the scope to creation and leave no space for situations where we're not creating instance but reusing already existing. In this case initialize parameters may be passed by a client (e.g. Theia) or received from the registry. Your suggestion fits current state but shortens the general flexibility.